crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -783,4 +786,16 @@ mod tests {
         schedule.run(&mut world);
         assert!(world.get_resource::<R>().is_some());
     }
+
+    /// Regression test for a weird stacked borrow bug flagged by MIRI in
+    /// `spawn_exclusive_system_task`, related to a `&mut World` being captures
+    /// inside an `async` block and somehow remaining alive even after its last use.
+    #[test]
+    fn check_spawn_exclusive_system_task_stacked_borrows() {
```

Comment:
I would remove naming stacked borrows specifically, since tree borrows hit this as well